### PR TITLE
I/6641 Table navigation using keyboard should not alter the native Shift+Arrow behavior

### LINKED ISSUE #6641: Table navigation using keyboard should not alter the native Shift+Arrow behavior
## 📝 Provide detailed reproduction steps (if any)

1. Go to a table cell.
2. Put selection in the middle of the text.
3. Shift+Arrow (e.g. down arrow)

## 📃 Other details

Follow-up of https://github.com/ckeditor/ckeditor5/issues/3267. Caused by https://github.com/ckeditor/ckeditor5-table/pull/304.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Table navigation using keyboard should not alter the native Shift+Arrow behavior inside table cells. Closes ckeditor/ckeditor5#6641.

---

### CODE REVIEW

**Review by jodator — COMMENTED** [review]
The code looks good. Waiting for more test and we can close this one as well.

File: tests/tablenavigation.js [hunk 1]
```diff
@@ -1714,6 +1714,8 @@ describe( 'TableNavigation', () => {
 						`
 					) );
 					global.document.querySelector( 'head' ).appendChild( styleElement );
+
+					editor.editing.view.focus();
```
Comment (jodator) [inline tests/tablenavigation.js:1718]: Could you add comment about this. I reckon that this cause problems but I don't remember what was exactly problematic here. Anyway maybe I'll learn why we do that :D

File: tests/tablenavigation.js [hunk 2]
```diff
@@ -1841,6 +1898,77 @@ describe( 'TableNavigation', () => {
...
+								[ '20', '21', '22' ]
+							] ) );
+
+							editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+							sinon.assert.notCalled( downArrowDomEvtDataStub.preventDefault );
+							sinon.assert.notCalled( downArrowDomEvtDataStub.stopPropagation );
+						} );
+
+						it( 'should expand selection to the beginning of the cell content', () => {
```
Comment (jodator) [inline tests/tablenavigation.js:1934]: I'd add "(collapsed selection)" and another test with "(expanded selection)" to test behavior on:

```
word[ word]' + tex
```

File: tests/tablenavigation.js [hunk 3]
```diff
@@ -1841,6 +1898,77 @@ describe( 'TableNavigation', () => {
...
+							sinon.assert.calledOnce( upArrowDomEvtDataStub.stopPropagation );
+
+							assertEqualMarkup( getModelData( model ), modelTable( [
+								[ '00', '01', '02' ],
+								[ '10', '[word] word' + text, '12' ],
+								[ '20', '21', '22' ]
+							] ) );
+						} );
+
+						it( 'should expand selection to the end of the cell content', () => {
```
Comment (jodator) [inline tests/tablenavigation.js:1953]: The same here for collapsed/expanded selection.

File: tests/tablenavigation.js [hunk 4]
```diff
@@ -1970,6 +2098,51 @@ describe( 'TableNavigation', () => {
...
 						] ) );
 					} );
+
+					describe( 'when shift key is pressed', () => {
+						beforeEach( () => {
+							upArrowDomEvtDataStub.shiftKey = true;
+							downArrowDomEvtDataStub.shiftKey = true;
+						} );
+
+						it( 'should expand selection to the beginning of the cell content', () => {
```
Comment (jodator) [inline tests/tablenavigation.js:2108]: Also for tests in wrap area - it would be nice to have also tests for expanded selection.

**Review by jodator — CHANGES_REQUESTED** [review]
I would add tests for expanded selection to have it defined and ready for any future changes :slightly\_smiling\_face:.

File: tests/tablenavigation.js (hunk 1 above)
Comment (niegowski) [reply tests/tablenavigation.js:1720]: Addressed in af0b27f.

File: tests/tablenavigation.js (hunk 4 above)
Comment (niegowski) [reply tests/tablenavigation.js:2193]: Adressed in fedf508.

**Review by jodator — APPROVED** [review]
LGMT :+1:

**Comment by niegowski** [comment]
Moved to ckeditor/ckeditor5#6727

**Pull request closed without merging**